brainiak/utils/fmrisim.py

```diff
@@ -1476,7 +1496,7 @@ def _generate_noise_temporal_autoregression(timepoints,
                     random = np.random.normal(0, 1)
                     temp.append(past_trs * past_reg + random)
 
-                    noise_autoregression.append(np.mean(temp))
+            noise_autoregression.append(np.mean(temp))
```

Comment:
I still think that there is some issue here by adding Gaussian noise multiple times and taking the mean of `temp`. I think if you generate AR(2) noise with the current equation and recover the rhos with functions in `nitime` package, you might find both the rhos and variance being half of the true values.
It should be done by *summing* the product of the past signals and their corresponding rho coefficients, and adding only the Gaussian noise only once, but no averaging. The way to generate first p (p being the order) sample is more complex for p>1.
I think this document is correct that you can refer to:
https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=5&ved=0ahUKEwjZ7bu-yLDXAhUCziYKHUwxAaQQFghAMAQ&url=http%3A%2F%2Fwww.stat.washington.edu%2Fcourses%2Fstat520%2Fwinter99%2Far-recipe%2Far-recipe.ps&usg=AOvVaw2wGB5KOz-LlYyNxTjLitFy

Reply:
Aha good find, I now see the problem. I did some texting with the updated code and it seems improved

Reply:
I can't find a way to comment on your above comment about downsampling the stimfunction for the convolution so here it is: I did infact mean 0.01ms. Since the HRF is scaled if an impulse occurs for any non zero time it will elicit the same HRF as an event lasting about 1s. This is a problem I don't think has any easy solution but I would love to hear

Reply:
Also won't let me reply to below: I am not sure what the current issues are with zscoring: at the moment it doesn't really matter that I do it because already the mean is ~0 and std is ~1 but I could remove it if you think it is necessary

Reply:
@CameronTEllis One thing I noticed is the default rho is 1. You probably do not want this default value because you will end up generating Brownian motion with the variance ever growing. Some number smaller than 1 is perhaps more sensible. Please also keep in mind the issue with AR(p) and fix it in future.

Reply:
@CameronTEllis, this seems to be the last sticking point.

Reply:
Changed the value to 0.5. It is a good compromise to avoid drift while still retaining autoregressive properties.